src/plugins/telemetry/server/plugin.ts

```diff
@@ -134,17 +127,26 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
 
     // If the opt-in selection cannot be changed, set it as early as possible.
     const { optIn, allowChangingOptInStatus } = this.initialConfig;
-    if (allowChangingOptInStatus === false) {
-      this.isOptedIn$.next(optIn);
-    }
+    this.isOptedIn = allowChangingOptInStatus === false ? optIn : undefined;
+
+    // Poll for the opt-in status
+    this.isOptedIn$ = timer(0, OPT_IN_POLL_INTERVAL_MS).pipe(
+      takeUntil(this.pluginStop$),
+      switchMap(() => this.getOptInStatus()),
```

Comment:
can this lead to memory leaks? 

If `getOptInStatus` takes longer than `OPT_IN_POLL_INTERVAL_MS` to execute, `switchMap` will trigger a new call on every tick of the `timer`.

Reply:
I switched back to `exhaustMap` and moved the `takeUntil` operator after the `exhaustMap`. This way, the RxJS pipeline seems to be `completed()` correctly.